packages/dd-trace/src/tracer.js

```diff
@@ -58,17 +58,25 @@ class DatadogTracer extends Tracer {
     }
   }
 
-  wrap (name, options, fn) {
+  wrap (name, options, fn, requiresParent = false) {
     const tracer = this
 
     return function () {
-      const cb = arguments[arguments.length - 1]
+      if (requiresParent && !tracer.scope().active()) {
```

Comment:
`tracer.trace` takes a function which may do things like modify arguments to the inner function that we're wrapping (to add its own callback, for example). It may also call a `done` function. If we want to bail out of tracing, we don't want any of that run, and we want to run the inner function directly. `tracer.trace` doesn't even actually take in a reference to the inner function, so there's nothing to call except the passed-in function. Because of all that, it made more sense to do this in `tracer.wrap`.

As an alternative, maybe the function passed to `tracer.trace` could switch its behavior based on whether `span` is provided, but then every such function would have to be modified to handle the tracing and non-tracing cases.

Reply:
I'm not sure I understand the limitation. When the option is used and `tracer.trace` is called, can't it simply not create the span, not touch the scope, and then done becomes a no-op?

Reply:
If it doesn't create the span, then what's being passed in as the first argument to the trace callback? null or undefined? Then any code interacting with `span` will have to fist check if it's present. I suppose that's not a big deal if they're already passing in the option, so they know that's a possibility.

Reply:
Yeah, I mean that's technically the whole point of the option. I guess it would have been more straightforward to not pass the span at all and rely on the scope manager.